octave/autotest.m

```diff
@@ -42,7 +44,6 @@ function plot_sig_and_error(plotnum, subplotnum, sig, error, titlestr, axisvec)
     axis(axisvec);
   end
   title(titlestr);
-  close(f, "force");
 endfunction
```

Comment:
Ooops I may have removed yr `close` @timw - however an immediate close will make the plots dissappear which isn't useful for real world use for debugging. These tests are used outside of cmake during development.

Hmm, so we need a way fwd that is OSX friendly and supports my regular work flow.

Reply:
I often use ```close all``` at the end of an Octave session, or to clean up plots.  That could be placed at the very end of the Octave script(s).

Reply:
I actually tried close all at the end before but I was still getting crashes because it looked like some supposedly created plots weren't actually getting created properly. I'm not sure why, unfortunately.

Would having them go directly to disk instead of the screen still allow your workflow?

Reply:
OK - this is a tricky one :thinking: 

A ```.png``` file might be acceptable - but we are still using the plot subsystem so might hit the same issue.  Do you have any more info on the tests that are not being created properly?

Reply:
@drowe67, two examples where this is happening are tnewamp1.m and tcohpsk.m. It seemed like the first plot wasn't displaying but the others were.

Reply:
@tmiw on OSX, could you pls try modifying the ctest command for one of those problem tests to be something like:
```
DISPLAY="" octave -qf "/name/of/octave/file"
```
This uses the text console as the graphics engine.

Reply:
@drowe67, I commented out the close calls in tnewamp1.m but still got a segfault with that command:

```
MacBook-Pro:octave mooneer$ DISPLAY="" octave -qf run_tnewamp1.m 
starting tnewamp1.c input_prefix: /Users/mooneer/devel/codec2/build_osx/src/cq_ref
warning: load: '/Users/mooneer/devel/codec2/octave/train_120_1.txt' found by searching load path
warning: called from
    tnewamp1 at line 72 column 3
    run_tnewamp1 at line 5 column 1
warning: load: '/Users/mooneer/devel/codec2/octave/train_120_2.txt' found by searching load path
warning: called from
    tnewamp1 at line 72 column 26
    run_tnewamp1 at line 5 column 1
Equaliser................: OK
rate_K_surface...........: OK
mean.....................: OK
rate_K_surface_..........: OK
interpolated_surface_....: OK
interpolated Wo_.........: OK
interpolated voicing.....: OK
rate L Am surface .......: OK
phase surface............: OK
passes: 9 fails: 0
warning: gh_manager::do_close_all_figures: some graphics elements failed to close.
fatal: caught signal Segmentation fault: 11 -- stopping myself...
Segmentation fault: 11
MacBook-Pro:octave mooneer$ 
```

However, I tried using octave-cli instead of octave and got no segfaults. Additionally, AquaTerm ran after test completion with only "Figure 1" and "Figure 2" displayed.

Reply:
@tmiw OK in https://github.com/drowe67/codec2/pull/108/commits/39cd03ab3c9f003273795c21f21cf38855802f5c I've tried the ```DISPLAY="" octave-cli``` approach for those problem tests, and also have a way to run ```tnewamp``` from the octave CLI during my development, and headless during ctests, by passing paths to the octave function on the command line.

Please let me know if that helps on OSX :smile:

Reply:
@drowe67, I had a few segfaults after that change so I just converted a few more to octave-cli and pushed those changes.

Reply:
@tmiw Travis seems happy!  Are you getting all passes on OSX now?

Reply:
@drowe67, yep, just ran it again and had 100% pass (minus the disabled Valgrind tests of course).